contracts/managers/AssetPairManagerV2.sol

```diff
@@ -0,0 +1,397 @@
+    {
+        rebalancingSetToken.setLiquidator(_newLiquidator);
+    }
+
+    /*
+     * Update liquidatorData used by Rebalancing Set.
+     *
+     * @param _newLiquidatorData      New Liquidator data
+     */
+    function setLiquidatorData(
+        bytes calldata _newLiquidatorData
+    )
+        external
+        onlyOwner
+    {
+        liquidatorData = _newLiquidatorData;
+    }
+
+    /**
+     * Allows the owner to update fees on the Set. Fee updates are timelocked.
+     *
+     * @param _newFeeCallData    Bytestring representing feeData to pass to fee calculator
+     */
+    function adjustFee(
+        bytes calldata _newFeeCallData
+    )
+        external
+        onlyOwner
+        timeLockUpgrade
+    {
+        rebalancingSetToken.adjustFee(_newFeeCallData);
+    }
+
+    /*
+     * Update fee recipient on the Set.
+     *
+     * @param _newFeeRecipient      Address of new fee recipient
+     */
+    function setFeeRecipient(
+        address _newFeeRecipient
+    )
+        external
+        onlyOwner
+    {
+        rebalancingSetToken.setFeeRecipient(_newFeeRecipient);
+    }
+
+    /*
+     * Function returning whether initialPropose can be called without revert
+     *
+     * @return       Whether initialPropose can be called without revert
+     */
+    function canInitialPropose()
+        external
+        view
+        returns (bool)
+    {
+        // If RebalancingSetToken in valid state and new allocation different from last known allocation
+        // then return true, else false
+        return rebalancingSetTokenInValidState()
+            && calculateBaseAssetAllocation() != baseAssetAllocation
+            && hasConfirmationWindowElapsed();
+    }
+
+    /*
+     * Function returning whether confirmPropose can be called without revert
+     *
+     * @return       Whether confirmPropose can be called without revert
+     */
+    function canConfirmPropose()
+        external
+        view
+        returns (bool)
+    {
+        // If RebalancingSetToken in valid state and new allocation different from last known allocation
+        // then return true, else false
+        return rebalancingSetTokenInValidState()
+            && calculateBaseAssetAllocation() != baseAssetAllocation
+            && inConfirmationWindow();
+    }
+
+    /* ============ Internal ============ */
+
+    /*
+     * Calculate base asset allocation given market conditions
+     *
+     * @return       New base asset allocation
+     */
+    function calculateBaseAssetAllocation()
+        internal
+        view
+        returns (uint256)
+    {
+        return trigger.isBullish() ? bullishBaseAssetAllocation : bearishBaseAssetAllocation;  
+    }
+
+     /*
+     * Function returning whether the rebalanceInterval has elapsed and then RebalancingSetToken is in 
+     * Default state
+     *
+     * @return       Whether RebalancingSetToken is in valid state for rebalance
+     */
+    function rebalancingSetTokenInValidState()
```

Comment:
One thing we could consider is getting rid of the `FlexibleTimingManagerLibrary` and just use this. Think the reason we didn't before was for more robust error logging but not sure that's worth it/necessary.

Reply:
This way sounds good to me